mmhuman3d/core/visualization/visualize_smpl.py

```diff
@@ -965,59 +997,55 @@ def render_smpl(
     # initialize the renderer.
     renderer = SMPLRenderer(
         resolution=render_resolution,
-        faces=faces,
         device=device,
         output_path=output_path,
-        colors=colors,
         return_tensor=return_tensor,
         alpha=alpha,
-        model_type=model_type,
-        img_format=img_format,
+        read_img_format=img_format,
         render_choice=render_choice,
-        projection=projection,
         frames_folder=frames_folder,
         plot_kps=plot_kps,
         vis_kp_index=vis_kp_index,
-        in_ndc=in_ndc,
         final_resolution=final_resolution,
         **render_param_dict)
 
-    renderer = renderer.to(device)
+    cameras = build_cameras(
+        dict(
+            type=projection,
+            in_ndc=in_ndc,
+            device=device,
+            K=K,
+            R=R,
+            T=T,
+            resolution=render_resolution))
 
+    if image_array is not None:
+        image_array = torch.Tensor(image_array)
+        image_array = align_input_to_padded(
+            image_array, ndim=4, batch_size=num_frames, padding_mode='ones')
     # prepare the render data.
-    render_dataset = RenderDataset(
-        **{
-            'images': image_array,
-            'vertices': vertices,
-            'K': K,
-            'R': R,
-            'T': T,
-            'joints': joints,
-            'joints_gt': kp3d,
-        })
-    RenderLoader = DataLoader(
-        dataset=render_dataset, batch_size=batch_size, shuffle=False)
-
-    # start rendering. no grad if non-differentiable render.
-    # return None if return_tensor is False.
-    results = []
-    for data in tqdm(RenderLoader):
-        if no_grad:
-            with torch.no_grad():
-                result = renderer(**data)
-                torch.cuda.empty_cache()
-                results.append(result)
-        else:
-            result = renderer(**data)
-            results.append(result)
-    renderer.export()
+    render_data = dict(
+        backgrounds=image_array,
```

Comment:
```suggestion
        images=image_array,
```
See `forward` method for details.

Reply:
ok